rust/src/lib/nm/settings/ip.rs

```diff
@@ -299,3 +337,18 @@ fn apply_nmstate_wait_ip(
         None => (),
     }
 }
+
+fn rules_contains_local_rule(rules: &[RouteRuleEntry], is_ipv6: bool) -> bool {
+    for rule in rules {
+        if let Some(family) = rule.family {
+            if is_ipv6 != matches!(family, AddressFamily::IPv6) {
+                continue;
+            }
+        }
+        if rule.table_id == Some(255) && rule.priority == Some(0) {
+            return true;
```

Comment:
This indicate we set `NM.replace_local_rule = false` when user state

```
route-rules:
  config:
    - family: ipv4
      route-table: 255
      priority: 0
      from: 192.168.1.1
```

Reply:
That is a nice catch. Thanks, I will fix that